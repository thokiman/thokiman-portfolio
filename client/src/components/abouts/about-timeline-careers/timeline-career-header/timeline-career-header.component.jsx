import React from 'react';
import TimelineCareerBelow from '../timeline-career-below/timeline-career-below.component';
import TimelineCareerUpper from '../timeline-career-upper/timeline-career-upper.component';
import {
  SeparatedLineTimelineHeader,
  TimelineHeaderContainer,
  TimelineLine,
  TimelineOverlay,
} from './timeline-career-header.styles';

export const TimelineCareerHeader = () => {
  return (
    <TimelineHeaderContainer className='timeline-header-container'>
      <TimelineCareerUpper />
      <SeparatedLineTimelineHeader className='timeline-header-separator'>
        <TimelineOverlay />
        <TimelineLine />
      </SeparatedLineTimelineHeader>
      <TimelineCareerBelow />
    </TimelineHeaderContainer>
  );
};

export default TimelineCareerHeader;
