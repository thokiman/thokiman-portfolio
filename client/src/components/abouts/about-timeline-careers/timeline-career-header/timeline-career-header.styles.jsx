import styled, { keyframes } from "styled-components";

export const TimelineHeaderContainer = styled.div`
  width: 100%;
  display: grid;
  grid-auto-columns: repeat(auto, 1fr);
  grid-auto-rows: repeat(3, 1fr);
  gap: 10px 0px;
  grid-template-areas:
    "upper-timeline-header-container upper-timeline-header-container upper-timeline-header-container upper-timeline-header-container"
    "separated-line-timeline-header separated-line-timeline-header separated-line-timeline-header separated-line-timeline-header"
    "below-timeline-header-container below-timeline-header-container below-timeline-header-container below-timeline-header-container";

  align-items: center;
  padding: 10px;
  @media only screen and (max-width: 800px) {
    position: relative;
    left: -10px;
  }
`;
export const SeparatedLineTimelineHeader = styled.div`
  width: 420%;
  height: 5px;
  overflow: hidden;
  position: relative;
  left: 120px;
  @media only screen and (max-width: 800px) {
    width: 90vw;
    left: 15px;
  }
`;
const timelineWipe = keyframes`
      to {
            transform: scaleX(0);
        }
`;
export const TimelineOverlay = styled.div`
  background: rgba(18, 129, 148, 0.5);
  border-radius: 20px 20px 20px 20px;
  height: 10px;
  position: absolute;
  top: -2px;
  width: 100%;
  animation: ${timelineWipe} 3s ease;
  animation-iteration-count: infinite;
  animation-direction: alternate;
  animation-timing-function: cubic-bezier(0.645, 0.045, 0.355, 1);
`;

export const TimelineLine = styled.div`
  border: 3px solid rgba(188, 207, 213, 0.4);
  border-radius: 10px 10px 10px 10px;
  backface-visibility: hidden;
`;
TimelineHeaderContainer.displayName = "TimelineHeaderContainer";
SeparatedLineTimelineHeader.displayName = "SeparatedLineTimelineHeader";
TimelineOverlay.displayName = "TimelineOverlay";
TimelineLine.displayName = "TimelineLine";
