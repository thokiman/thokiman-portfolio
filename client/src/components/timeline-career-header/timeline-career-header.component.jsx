import React from "react";

import {
  TimelineHeaderContainer,
  SeparatedLineTimelineHeader,
  TimelineLine,
  TimelineOverlay,
} from "./timeline-career-header.styles";
import TimelineCareerUpper from "../timeline-career-upper/timeline-career-upper.component";
import TimelineCareerBelow from "../timeline-career-below/timeline-career-below.component";

const TimelineCareerHeader = () => {
  return (
    <TimelineHeaderContainer>
      <TimelineCareerUpper />
      <SeparatedLineTimelineHeader>
        <TimelineOverlay />
        <TimelineLine />
      </SeparatedLineTimelineHeader>
      <TimelineCareerBelow />
    </TimelineHeaderContainer>
  );
};

export default TimelineCareerHeader;
