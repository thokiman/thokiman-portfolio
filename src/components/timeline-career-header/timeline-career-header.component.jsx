import React from "react";

import TimelineCareerUpper from "../timeline-career-upper/timeline-career-upper.component";
import TimelineCareerBelow from "../timeline-career-below/timeline-career-below.component";
import "./timeline-career-header.styles.scss";

const TimelineCareerHeader = () => {
  return (
    <div className="timeline-header-container">
      <TimelineCareerUpper />
      <div className="separated-line-timeline-header" id="straigh-line">
        <div className="overlay"></div>
        <div className="line"></div>
      </div>
      <TimelineCareerBelow />
    </div>
  );
};

export default TimelineCareerHeader;
