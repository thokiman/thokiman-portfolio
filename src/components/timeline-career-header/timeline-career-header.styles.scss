.timeline-header-container {
    display: grid;
    grid-auto-columns: repeat(auto, 1fr);
    grid-auto-rows: repeat(3, 1fr);
    gap: 10px 0px;
    grid-template-areas:
        "upper-timeline-header-container upper-timeline-header-container upper-timeline-header-container upper-timeline-header-container"
        "separated-line-timeline-header separated-line-timeline-header separated-line-timeline-header separated-line-timeline-header"
        "below-timeline-header-container below-timeline-header-container below-timeline-header-container below-timeline-header-container";

    align-items: center;

    .separated-line-timeline-header {
        width: 420%;
        height: 5px;
        overflow: hidden;
        position: relative;
        left: 120px;

        .overlay {
            background: rgba($color:#128194, $alpha: 0.5);
            border-radius: 20px 20px 20px 20px;
            height: 10px;
            position: absolute;
            top: -2px;
            width: 100%;
            animation: wipe 3s ease;
            animation-iteration-count: infinite;
            animation-direction: alternate;
            animation-timing-function: cubic-bezier(0.645, 0.045, 0.355, 1.000);

        }

        .line {
            border: 3px solid rgba($color:#bccfd5, $alpha: 0.4);
            border-radius: 10px 10px 10px 10px;
            backface-visibility: hidden;
        }
    }

    @keyframes wipe {
        to {
            transform: scaleX(0);
        }
    }


    @keyframes pulsate-bck {
        0% {
            transform: scale(1);
        }

        50% {
            transform: scale(0.9);
        }

        100% {
            transform: scale(1);
        }
    }

    @keyframes rotate-90-bl-ccw {
        0% {
            transform: rotate(0);
            transform-origin: 0% 100%;
        }

        100% {
            transform: rotate(-90deg);
            transform-origin: 0% 100%;
        }
    }
}

@keyframes rotate-90-tr-ccw {
    0% {
        transform: rotate(0);
        transform-origin: top right;
    }

    100% {
        transform: rotate(-90deg);
        transform-origin: top right;
    }
}

@keyframes color-change-2x {
    0% {
        color: rgba($color: #128194, $alpha: 0.6);
    }

    100% {
        color: rgba($color:#3f4249, $alpha: 0.9);
    }
}

@keyframes swing-in-bottom-bck {
    0% {
        transform: rotateX(-70deg);
        transform-origin: bottom;
        opacity: 0;
    }

    100% {
        transform: rotateX(0);
        transform-origin: bottom;

        opacity: 1;
    }
}

@keyframes swing-in-top-bck {
    0% {
        transform: rotateX(70deg);
        transform-origin: top;
        opacity: 0;
    }

    100% {
        transform: rotateX(0deg);
        transform-origin: top;
        opacity: 1;
    }
}