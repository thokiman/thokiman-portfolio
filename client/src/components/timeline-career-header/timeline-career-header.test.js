import React from "react";
import { shallow } from "enzyme";

import { TimelineCareerHeader } from "./timeline-career-header.component";

describe("TimelineCareerHeader component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TimelineCareerHeader />);
  });

  it("should render TimelineCareerHeader component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
