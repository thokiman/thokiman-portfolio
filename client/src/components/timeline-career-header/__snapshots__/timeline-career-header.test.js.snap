// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineCareerHeader component should render TimelineCareerHeader component 1`] = `ShallowWrapper {}`;
